### PR TITLE
Fix visualization docs

### PR DESCRIPTION
**Pull request by nonhermitian**
### Summary
The visualization docs were being built twice due to the switch to module level doc strings in source.  Additionally `autosummary` was not doing its thing because it was not setup correctly.

### CODE REVIEW

File: docs/apidoc/qiskit.rst [hunk 1]
```diff
@@ -25,4 +25,4 @@ API Reference
    tools/tools
    transpiler/transpiler
    validation
-   visualization/visualization
+   visualization
```
Comment (mtreinish) [inline docs/apidoc/qiskit.rst:28]: Should we remove the docs/apidoc/visualization directory too?

**Review by mtreinish — COMMENTED** [review]
This LGTM. I noticed an issue with a stray `)` at the end of visualization on the side bar:

Screenshot_2019-10-24_08-50-49

But it looks like that's likely an issue with the terra source since it's there currently just one level lower (because of the duplication this fixes)

One inline question then I think this is good to go

File: docs/apidoc/qiskit.rst (hunk 1 above)
Comment (nonhermitian) [reply docs/apidoc/qiskit.rst:28]: The dir should be removed and replaced with just a `visualization.rst` file with:

```

.. _qiskit-visualization:

.. automodule:: qiskit.visualization
   :no-members:
   :no-inherited-members:
   :no-special-members
```

Comment (mtreinish) [reply docs/apidoc/qiskit.rst:28]: oh, I just meant the PR still left the visualization dir intact. I've updated it here: https://github.com/Qiskit/qiskit/pull/642/commits/4195ad0ab8ed3d3779ee9d6300927aab3ac4071f

**Review by mtreinish — APPROVED** [review]

**Pull request merged**